include/swift/AST/ASTPrinter.h

```diff
@@ -136,7 +139,8 @@ class ASTPrinter {
   /// \param T the original \c Type being referenced. May be null.
   /// \param RefTo the \c TypeDecl this is considered a reference to.
   /// \param Name the name to be printed.
```

Comment:
This needs to be updated to document the new parameter.

I'm also not 100% happy with the name `printedModuleContext`. After reading the whole diff, I now understand that it indicates that you've already printed a module name and you're chaining from it, but it didn't read that way to me immediately. Maybe adding the doc comment would suffice, but maybe something like `inModuleContext` or just `alreadyPrintedModuleContext` would be a bit clearer.

On the other hand, maybe the flag should be tweaked to indicate "is qualified"/"not qualified" (and also pass `true` after chaining from parent type to child type), or maybe...

[continued in `ASTPrinter::printTypeRef()` annotation]